test/e2e_node/e2e_service.go

```diff
@@ -208,6 +202,45 @@ func (es *e2eService) startServer(cmd *healthCheckCommand) error {
 	return fmt.Errorf("Timeout waiting for service %s", cmd)
 }
 
+func (es *e2eService) stopService(name string) error {
+	// Since node services are started with sudo, we must get the PID of the child process.
+	out, err := exec.Command("pidof", name).Output()
```

Comment:
Would setting `exec.Cmd.SysProcAttr.Pdeathsig` while execin'g the processes
solve this problem?

On Fri, Jun 3, 2016 at 1:56 PM, Tim St. Clair <notifications@github.com>
wrote:

> In test/e2e_node/e2e_service.go
> <https://github.com/kubernetes/kubernetes/pull/26735#discussion_r65774195>
> :
>
> > @@ -208,6 +202,45 @@ func (es *e2eService) startServer(cmd *healthCheckCommand) error {
> >  	return fmt.Errorf("Timeout waiting for service %s", cmd)
> >  }
> >
> > +func (es *e2eService) stopService(name string) error {
> > +	// Since node services are started with sudo, we must get the PID of the child process.
> > +	out, err := exec.Command("pidof", name).Output()
>
> Using - only works on process groups (pgid). I'm not sure, but I think
> that might be the same as the test framework?
>
> stclair@node-e2e:~$ ps -eo pid,pgid,argsess h for help or q to quit)  PID  PGID COMMAND... 9766  9729 sudo /tmp/gcloud-e2e-1774959747/kube-apiserver --etcd-servers http://127.0.0.1:4001 --insecure-bind-address 0.0.0.0 --service-cluster-ip-range 10.0.0.1/24 --kubelet-port 10250 --allow-privileged true --v 8 --logtostderr 9767  9729 /tmp/gcloud-e2e-1774959747/kube-apiserver --etcd-servers http://127.0.0.1:4001 --insecure-bind-address 0.0.0.0 --service-cluster-ip-range 10.0.0.1/24 --kubelet-port 10250 --allow-privileged true --v 8 --logtostderr 9771  9729 sudo /tmp/gcloud-e2e-1774959747/kubelet --api-servers http://127.0.0.1:8080 --address 0.0.0.0 --port 10250 --hostname-override node-e2e.us-central1-a.stclair-k8s-ext --volume-stats-agg-period 10s --allow-privileged true --ser 9774  9729 /tmp/gcloud-e2e-1774959747/kubelet --api-servers http://127.0.0.1:8080 --address 0.0.0.0 --port 10250 --hostname-override node-e2e.us-central1-a.stclair-k8s-ext --volume-stats-agg-period 10s --allow-privileged true --serializ 9783  9729 journalctl -k -f...
>
> —
> You are receiving this because you were assigned.
> Reply to this email directly, view it on GitHub
> <https://github.com/kubernetes/kubernetes/pull/26735/files/cacd65e9f2131f4a87fc7b4ec7c3a2c4ba3c97f8#r65774195>,
> or mute the thread
> <https://github.com/notifications/unsubscribe/AGvIKBwIiIhjddMILhRaAgyvlE9zG8tEks5qIJUTgaJpZM4Is8Sh>
> .
>